### PR TITLE
More modern CMake and cleanup

### PR DESCRIPTION
**Pull request by IohannRabeson**
Hi! I made those changes on my own fork and you might be interested.
 - I'm using more modern CMake when possible. This reduce CMake code to write when using Mustache and CMake, just have to specify Mustache as dependency and the correct directory should be known and C++17 is enabled as well.
 - I renamed the executable generated by the test target to make it obvious this executable is a test.

### CODE REVIEW

**Comment by kainjow** [comment]
I would like to keep Travis tests for now.

**Comment by IohannRabeson** [comment]
No problems, I will bring it back.

**Comment by IohannRabeson** [comment]
Let me know if you need anything else.

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -1,21 +1,12 @@
...
+    target_compile_options(mustache-unit-tests PRIVATE -Wall -Wextra -Werror)
 elseif (MSVC)
-	add_definitions(
-		/W3
-		/WX
-	)
+    target_compile_options(mustache-unit-tests PRIVATE /W3 /WX)
 endif()
-add_executable(mustache
-	../mustache.hpp # to show in IDE
```
Comment (kainjow) [inline tests/CMakeLists.txt:17]: oops, I realized this is useful to have these sources show up in the IDE, such as Xcode. Can you add them back in a new PR?

Comment (IohannRabeson) [reply tests/CMakeLists.txt:17]: Yes I will do that

**Pull request merged**